### PR TITLE
Fixes to issue #610.

### PR DESCRIPTION
**Pull request by alkrall**
Fixes to issue #610

### CODE REVIEW

File: R/importWQP.R [hunk 1]
```diff
@@ -145,6 +151,9 @@ importWQP <- function(obs_url, zip=TRUE, tz="UTC",
       attr(emptyReturn$ActivityStartDateTime, "tzone") <- tz
       attr(emptyReturn$ActivityEndDateTime, "tzone") <- tz
       
+      if(!convertType){
```
Comment (ldecicco-USGS) [inline R/importWQP.R:154]: I guess if convertType is true, we would want to make this empty data frame completely character

File: R/whatWQPdata.R [hunk 2]
```diff
@@ -31,7 +31,7 @@ whatWQPsamples <- function(...){
   baseURL <- appendDrURL(baseURL, mimeType = "tsv")
   
   withCallingHandlers({
-    retval <- importWQP(baseURL, zip=values["zip"] == "yes")
+    retval <- importWQP(baseURL, zip=values["zip"] == "yes", convertType=convertType)
```
Comment (ldecicco-USGS) [inline R/whatWQPdata.R:34]: I don't know where this convertType value would come from. I think we want to add it to the function itself (?). So something like this:
```r
whatWQPsamples <- function(..., convertType = TRUE){

```
Then you'd need to add the @param stuff to this function.

File: R/whatWQPdata.R [hunk 3]
```diff
@@ -74,7 +76,7 @@ whatWQPmetrics <- function(...){
...
```
Comment (ldecicco-USGS) [inline R/whatWQPdata.R:79]: same as whatWQPsamples above

**Comment by ldecicco-USGS** [comment]
When you click "Install" from the Build tab, you should be updating the .Rd files. If not, in that same Build tab, click More -> Configure Build Tools -> then click the checkbox for "Generate documentation with Roxygen". 

Once that's done, you should see a few .Rd files that need to be committed and then Pushed up to this PR. The R Check that is giving some Warnings are saying the help files need some information on this new argument "convertType". I see you updated the Roxygen code above the functions themselves, but we need that Roxygen code to actually update the .Rd files in the man folder.

File: R/importWQP.R (hunk 1 above)
Comment (alkrall) [reply R/importWQP.R:154]: Updated all columns to character data type within the  'emptyReturn' data frame.

**Comment by alkrall** [comment]
Hi Laura,
Let me know if I addressed these correctly.

Aliesha L. Krall
Physical Scientist
U.S. Geological Survey Upper Midwest Water Science Center
2280 Woodale Drive
Mounds View, MN 55112
office: 763-783-3135
https://www.usgs.gov/staff-profiles/aliesha-l-diekoff
________________________________
From: Laura DeCicco ***@***.***>
Sent: Monday, August 22, 2022 1:12 PM
To: USGS-R/dataRetrieval ***@***.***>
Cc: Krall, Aliesha L ***@***.***>; Author ***@***.***>
Subject: [EXTERNAL] Re: [USGS-R/dataRetrieval] Fixes to issue #610. (PR #632)




 This email has been received from outside of DOI - Use caution before clicking on links, opening attachments, or responding.



When you click "Install" from the Build tab, you should be updating the .Rd files. If not, in that same Build tab, click More -> Configure Build Tools -> then click the checkbox for "Generate documentation with Roxygen".

Once that's done, you should see a few .Rd files that need to be committed and then Pushed up to this PR. The R Check that is giving some Warnings are saying the help files need some information on this new argument "convertType". I see you updated the Roxygen code above the functions themselves, but we need that Roxygen code to actually update the .Rd files in the man folder.

File: R/importWQP.R (hunk 1 above)
Comment (ldecicco-USGS) [reply R/importWQP.R:154]: We don't want to make everything character unless the user choose convertType = FALSE. So we'll need to get the emptyReturn back to it's original types (numerics, dates, etc).

Then, you can add something like:
```r
if(!convertType){
  i <- sapply(emptyReturn, !is.character)
  emptyReturn[i] <- lapply(emptyReturn[i], as.character)
}
```
(you'll need to double-check that code....)

File: tests/testthat/tests_general.R [hunk 4]
```diff
@@ -556,3 +556,21 @@ test_that("readWQPsummary", {
...
+  
+  pH <- readWQPdata(statecode="WI", countycode="Dane", characteristicName="pH", convertType=FALSE)
+  expect_is(pH$ResultMeasureValue, "character")
+  
+  SC<-readWQPqw(siteNumbers="USGS-05288705", parameterCd="00300", convertType=FALSE)
+  expect_is(SC$ResultMeasureValue, "character")
+  
+  lakeSites_chars <- whatWQPdata(siteType = "Lake, Reservoir, Impoundment", 
+                                 statecode = "US:55", convertType=FALSE)
+  expect_is(lakeSites_chars$lat, "character")
```
Comment (ldecicco-USGS) [inline tests/testthat/tests_general.R:574]: All the github tests are failing here, they're showing that the lat column is coming in as numeric. Could you check on your end if you also get a numeric?

**Comment by ldecicco-USGS** [comment]
We're going pull this, and either myself or Aliesha will fix the broken tests.

**Pull request merged**